internal/testfixtures/validating.go

```diff
@@ -64,6 +64,25 @@ func (vsr validatingSnapshotReader) ListNamespaces(
 	return read, err
 }
 
+func (vsr validatingSnapshotReader) LookupNamespaces(
+	ctx context.Context,
+	nsNames []string,
+) ([]*core.NamespaceDefinition, error) {
+	read, err := vsr.delegate.LookupNamespaces(ctx, nsNames)
+	if err != nil {
+		return read, err
```

Comment:
nit: may confuse the reader, even though `read` should be `nil`
```suggestion
		return nil, err
```